x-pack/plugins/security_solution/cypress/integration/detection_rules/indicator_match_rule.spec.ts

```diff
@@ -92,104 +110,337 @@ describe('Detection rules, Indicator Match', () => {
+      });
+
+      it('Shows invalidation text when the "AND" button is pressed and both the mappings are blank', () => {
+        getIndicatorAndButton().click();
+        getIndicatorInvalidationText().should('exist');
+      });
+
+      it('Shows invalidation text when the "OR" button is pressed and both the mappings are blank', () => {
+        getIndicatorOrButton().click();
+        getIndicatorInvalidationText().should('exist');
+      });
+
+      it('Does NOT show invalidation text when there is a valid "index field" and a valid "indicator index field"', () => {
+        fillIndicatorMatchRow({
+          indexField: newThreatIndicatorRule.indicatorMapping,
+          indicatorIndexField: newThreatIndicatorRule.indicatorIndexField,
+        });
+        getDefineContinueButton().click();
+        getIndicatorInvalidationText().should('not.exist');
+      });
+
+      it('Shows invalidation text when there is an invalid "index field" and a valid "indicator index field"', () => {
+        fillIndicatorMatchRow({
+          indexField: 'non-existent-value',
+          indicatorIndexField: newThreatIndicatorRule.indicatorIndexField,
+          validColumns: 'indicatorField',
+        });
+        getDefineContinueButton().click();
+        getIndicatorInvalidationText().should('exist');
+      });
+
+      it('Shows invalidation text when there is a valid "index field" and an invalid "indicator index field"', () => {
+        fillIndicatorMatchRow({
+          indexField: newThreatIndicatorRule.indicatorMapping,
+          indicatorIndexField: 'non-existent-value',
+          validColumns: 'indexField',
+        });
+        getDefineContinueButton().click();
+        getIndicatorInvalidationText().should('exist');
+      });
+
+      it('Deletes the first row when you have two rows. Both rows valid rows of "index fields" and valid "indicator index fields". The second row should become the first row', () => {
+        fillIndicatorMatchRow({
+          indexField: newThreatIndicatorRule.indicatorMapping,
+          indicatorIndexField: newThreatIndicatorRule.indicatorIndexField,
+        });
+        getIndicatorAndButton().click();
+        fillIndicatorMatchRow({
+          rowNumber: 2,
+          indexField: 'agent.name',
+          indicatorIndexField: newThreatIndicatorRule.indicatorIndexField,
+          validColumns: 'indicatorField',
+        });
+        getIndicatorDeleteButton().click();
+        getIndicatorIndexComboField().should('have.text', 'agent.name');
+        getIndicatorMappingComboField().should(
+          'have.text',
+          newThreatIndicatorRule.indicatorIndexField
+        );
+        getIndicatorIndexComboField(2).should('not.exist');
+        getIndicatorMappingComboField(2).should('not.exist');
+      });
+
+      it('Deletes the first row when you have two rows. Both rows have valid "index fields" and invalid "indicator index fields". The second row should become the first row', () => {
```

Comment:
I'm curious of the rationale behind these being cypress tests as opposed to integration-y jest tests with enzyme. It doesn't look like we're testing the e.g. the field autocompletion for many of these, but we're still paying the cost of loading/unloading data, page loads, etc.

Reply:
The bugs are around the DOM and seemed to make the most sense to do with Cypress (to me). I don't know if enzyme would show the same results/bugs around it since I'm trying to test the contents of the DOM. It could though. I also thought it would just be better to have my Cypress e2e tests from our last meeting but Enzyme might work out as well.

If these prove problematic for e2e tests due to flake I could try to move them to enzyme and see if I can get the same consistent DOM tests. You bring up a good point.

Reply:
Agreed, just something to keep an eye on!